Formula/llvm.rb

```diff
@@ -95,46 +120,60 @@ class Llvm < Formula
       url "http://llvm.org/git/lldb.git"
     end
 
-    # Polly is --HEAD-only because it requires isl and the version of Polly
-    # shipped with 3.6.2 only compiles with isl 0.14 and earlier (current
-    # version is 0.15). isl is distributed with the Polly source code from LLVM
-    # 3.7 and up, so --HEAD builds do not need to depend on homebrew isl.
-    option "with-polly", "Build with the experimental Polly optimizer"
     resource "polly" do
       url "http://llvm.org/git/polly.git"
     end
   end
 
   keg_only :provided_by_osx
 
+  # Options
   option :universal
-  option "with-clang", "Build the Clang compiler and support libraries"
+  option "without-clang", "Do not build the Clang compiler and support libraries"
   option "with-clang-extra-tools", "Build extra tools for Clang"
   option "with-compiler-rt", "Build Clang runtime support libraries for code sanitizers, builtins, and profiling"
-  option "with-libcxx", "Build the libc++ standard library"
+  option "without-libcxx", "Do not build the libc++ standard library"
+  option "with-libcxxabi", "Build the libc++abi standard library"
+  option "with-libunwind", "Build the libunwind library"
   option "with-lld", "Build LLD linker"
   option "with-lldb", "Build LLDB debugger"
   option "with-python", "Build Python bindings against Homebrew Python"
-  option "with-rtti", "Build with C++ RTTI"
+  option "without-rtti", "Do not build C++ RTTI"
   option "with-utils", "Install utility binaries"
-
-  deprecated_option "rtti" => "with-rtti"
-
-  if MacOS.version <= :snow_leopard
+  option "with-polly", "Build with the experimental Polly optimizer"
+  option "with-test", "Build LLVM unit tests"
+  option "with-shared-libs", "Build all libs as shared instead of static"
+  option "with-libffi", "Use libffi to call external functions from the interpreter"
+
+  # Dependencies
+  depends_on "libffi"   => :optional # llvm.org/docs/GettingStarted.grml#requirements
+  depends_on "doxygen"  => :optional # for C++ API reference (lldb)
+  depends_on "graphviz" => :optional # for the 'dot' tool (lldb)
+  depends_on "ocaml"    => :optional
+
+  # Python
+  if build.with? "python"
+    depends_on "python"
+  elsif MacOS.version <= :snow_leopard
     depends_on :python
   else
-    depends_on :python => :optional
+    depends_on :python => :optional # not sure this is correct
```

Comment:
Python is definitely needed outside the test. Also the Python binding will remember the Python interpreter, so we offer. `with-python` to build binding against non-system Python. Therefore, it should be:

```ruby
if MacOS.version <= :snow_leopard
  depends_on :python
else
  depends_on :python => :optional
end
```